aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/ObserveQueryExecutorTest.java

```diff
@@ -264,4 +263,140 @@ public void observeQueryCancelsTheOperationOnQueryError() throws DataStoreExcept
+                PublishSubject.<StorageItemChange<? extends Model>>create().toSerialized();
+        Consumer<DataStoreQuerySnapshot<BlogOwner>> onQuerySnapshot = value -> {
+            if (latch.getCount() > 0) {
+                Assert.assertTrue(value.getItems().contains(blogOwner));
+                latch.countDown();
+            } else if (latch.getCount() == 0) {
+                Assert.assertFalse(value.getItems().contains(blogOwner));
+                changeLatch.countDown();
+            }
+        };
+        Consumer<Cancelable> observationStarted = value -> {
+            try {
+                subject.onNext(StorageItemChange.<BlogOwner>builder()
+                        .changeId(UUID.randomUUID().toString())
+                        .initiator(StorageItemChange.Initiator.SYNC_ENGINE)
+                        .item(blogOwner)
+                        .patchItem(SerializedModel.create(blogOwner,
+                                ModelSchema.fromModelClass(BlogOwner.class)))
+                        .modelSchema(ModelSchema.fromModelClass(BlogOwner.class))
+                        .predicate(QueryPredicates.all())
+                        .type(StorageItemChange.Type.DELETE)
+                        .build());
+            } catch (AmplifyException exception) {
+                exception.printStackTrace();
+            }
+        };
+        Consumer<DataStoreException> onObservationError = NoOpConsumer.create();
+        Action onObservationComplete = () -> { };
+        SqlQueryProcessor mockSqlQueryProcessor = mock(SqlQueryProcessor.class);
+        when(mockSqlQueryProcessor.queryOfflineData(eq(BlogOwner.class), any(), any()))
+                .thenReturn(datastoreResultList);
+        when(mockSqlQueryProcessor.modelExists(any(), any())).thenReturn(false);
+        ExecutorService threadPool = Executors.newFixedThreadPool(
+                Runtime.getRuntime().availableProcessors() * 5);
+        ObserveQueryExecutor<BlogOwner> observeQueryExecutor = new ObserveQueryExecutor<>(subject,
+                mockSqlQueryProcessor,
+                threadPool,
+                mockSyncStatus,
+                new ModelSorter<>(),
+                maxRecords, 2);
+        observeQueryExecutor.observeQuery(
+                BlogOwner.class,
+                new ObserveQueryOptions(), observationStarted,
+                onQuerySnapshot,
+                onObservationError,
+                onObservationComplete);
+        Assert.assertTrue(latch.await(1, TimeUnit.SECONDS));
+        Assert.assertTrue(changeLatch.await(10, TimeUnit.SECONDS));
+    }
+
+    /***
+     * Test should Fail when ObserveQuery returns deleted record.
+     * @throws InterruptedException InterruptedException
+     * @throws DataStoreException DataStoreException
+     */
+    @Test(expected = AssertionError.class)
+    public void observeQueryShouldFailIfReturnsDeletedRecord() throws InterruptedException, DataStoreException {
```

Comment:
What is the purpose of this test? Why would observe query return a deleted record with the fix in place? If were forcing the wrong behavior in a test, this isn't a valid test.

Reply:
I have removed this test as it was redundant and testing the same behavior.